apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/PgNumericTests.cs

```diff
@@ -329,56 +330,104 @@ public void FromDecimal(decimal input, string expectedValue)
         {
-            // Decimal can take upto 29 digits. If input has more significant digits than decimal can handle, it will be truncated and
-            // there will be loss of precision.
-            var numeric = PgNumeric.Parse(input);
-            var convertedDecimalString = numeric.ToDecimal(LossOfPrecisionHandling.Truncate).ToString(CultureInfo.InvariantCulture);
-            Assert.Equal(expectedValue, convertedDecimalString);
+            // It's easier to do this than to vary negation in other ways.
+            PerformTest(text, expectedDecimalText);
+            PerformTest("-" + text, "-" + expectedDecimalText);
+
+            static void PerformTest(string text, string expectedDecimalText)
+            {
+                PgNumeric numeric = PgNumeric.Parse(text);
+                decimal actual = numeric.ToDecimal(LossOfPrecisionHandling.Truncate);
+                Assert.Equal(expectedDecimalText, actual.ToString(CultureInfo.InvariantCulture));
+                // Conversion via the explicit conversion should do the same thing
+                decimal actual2 = (decimal)numeric;
+                Assert.Equal(actual, actual2);
+            }
         }
 
-        [Fact]
-        public void ToDecimal_Overflow()
+        [Theory]
+        [InlineData("0.00000000000000000000000000000000000000", "0.0000000000000000000000000000")]
+        [InlineData("-0.00000000000000000000000000000000000000", "0.0000000000000000000000000000")]
+        public void ToDecimal_LossOfPrecision_Truncate_Zero(string text, string expectedDecimalText)
         {
-            var numeric = PgNumeric.Parse("99999999999999999999999999999999.9999999999999999999999999999");
-            Assert.Throws<OverflowException>(() => numeric.ToDecimal(LossOfPrecisionHandling.Truncate));
+            PgNumeric numeric = PgNumeric.Parse(text);
+            decimal actual = numeric.ToDecimal(LossOfPrecisionHandling.Truncate);
+            Assert.Equal(expectedDecimalText, actual.ToString(CultureInfo.InvariantCulture));
+            // Conversion via the explicit conversion should do the same thing
+            decimal actual2 = (decimal)numeric;
+            Assert.Equal(actual, actual2);
         }
 
-        [Theory]
-        [InlineData("12345678.1234567891234567891234567891")]
-        [InlineData("123456789012345678900.123456789")]
-        public void ToDecimal_LossOfPrecision_Throw(string input)
+        [Theory, CombinatorialData]
+        public void ToDecimal_LossOfPrecision_Throw(
+            [CombinatorialValues(
+                "79228162514264337593543950335.00000000000000000000000000000000000001", // decimal.MaxValue + epsilon; doesn't count as overflow
+                "123456789012345678900.123456789", // Simpler example of more significant digits than decimal can handle
+                "0.00000000000000000000000000000000000000" // Even for zero, this loses information as there are more significant digits than decimal can handle
```

Comment:
Please note this test case, as I believe this is right, i.e. that we are losing information here even if for zero, because this number is not fixed point.

Reply:
Hmm... I would disagree here. There are more *specified* digits, but they're not *significant* in the normal mathematical sense. Basically I think we should be okay to lose trailing zeroes after a decimal point. On the other hand, as we do preserve them normally, we're definitely losing *information* - just not significant digits :)

I think it's okay to keep the behavior, but let's document it and change the comment.